R/data.table.R

```diff
@@ -1892,7 +1892,10 @@ as.matrix.data.table = function(x, rownames=NULL, rownames.value=NULL, ...) {
   }
   if (!is.null(rownames)) {
     # extract that column.
-    rownames.value = x[[rownames]]
+    if (is.ff(x[[rownames]]))
+      rownames.value = x[[rownames]][] # bring the column into memory if ff object
```

Comment:
can we actually really test that without adding `ff` to suggests?

Reply:
A quick scan of tests.Rraw, I can't see any tests that covered the per-column check for `ff` that was previously in the existing `as.matrix.data.table` function. For the purposes of codecov we could cheat in the same way as the `is.ff()` function does: simply add the `ff` class to a column in the test so that it hits those checks (although obviously this would not test the pulling into memory functionality)